rama-http/src/service/web/router.rs

```diff
@@ -100,7 +49,125 @@ where
     /// create a new web router
     pub(crate) fn new() -> Self {
         Self {
-            routes: TrieNode::new(),
+            routes: MatchitRouter::new(),
+            not_found: Arc::new(
```

Comment:
this can probably be made optional so you don't need to assign it if not set, and instead just fallback to NOT_FOUND anyway but on the spot